### PR TITLE
Ontology Box init by top left point and bottom right point

### PR DESCRIPTION
**Pull request by hepengfe**
This PR fixes  https://github.com/asyml/forte/issues/878.

### Description of changes

* The default Box init method: height and width and center -> init Box by top left point and bottom right point
* move init by  height &  width & center to a classmethod

### Possible influences of this PR.
It can simplify the Box API and enable two ways of initialization.

### Test Conducted
Tested wrong box shape and wrong x/y coordinates of top left point and bottom right point.

### CODE REVIEW

File: forte/data/ontology/top.py [hunk 1]
```diff
@@ -1105,9 +1190,10 @@ def corners(self):
...
-            (self._cy + h_offset, self._cx + w_offset)
-            for h_offset in [-0.5 * self._height, 0.5 * self._height]
-            for w_offset in [-0.5 * self._width, 0.5 * self._width]
+            (self.y0, self.x0),
+            (self.y0, self.x1),
+            (self.y1, self.x0),
+            (self.y1, self.x1),
         ]
 
     @property
```
Comment (mylibrar) [inline forte/data/ontology/top.py:1199]: Is `box_min_x` just an equivalent of `max(self.x0, 0)`? Could you explain a little bit on why we need `box_min_x/y` and `box_max_x/y`? I think maybe we can compute IOU or check overlapping directly through `self.x0/1` and `self.y0/1`?

File: forte/data/ontology/top.py [hunk 2]
```diff
@@ -1189,42 +1250,30 @@ class BoundingBox(Box):
     the image/grid.
 
     Args:
-        pack: The container that this BoundingBox will
-            be added to.
+        pack: the container that this ``Box`` will be added to.
```
Comment (mylibrar) [inline forte/data/ontology/top.py:1253]: This should be ``BoundingBox``

File: forte/data/ontology/top.py [hunk 3]
```diff
@@ -1189,42 +1250,30 @@ class BoundingBox(Box):
...
-        grid_cell_w_idx: the width index of the associated grid cell in
-            the grid, the unit is one grid cell.
-
+            it defaults to 0 which meaning it will load the first image payload.
     """
 
-    def __init__(
-        self,
+    @classmethod
+    def init_from_center_n_shape(
```
Comment (mylibrar) [inline forte/data/ontology/top.py:1264]: What's the difference between `BoundingBox.init_from_center_n_shape` and `Box.init_from_center_n_shape`? If they are exactly the same, then why do we need `BoundingBox` for this PR?

File: tests/forte/image_annotation_test.py [hunk 4]
```diff
@@ -56,3 +56,41 @@ def test_image_annotation(self):
...
+
+        self.assertRaises(ValueError, wrong_box)
+
+        def wrong_box():
+            # negative height
+            Box(self.datapack, [2, 2], [2, 1], 0)
+
+        self.assertRaises(ValueError, wrong_box)
+
+        b3 = BoundingBox.init_from_center_n_shape(self.datapack, 4, 4, 5, 5)
```
Comment (mylibrar) [inline tests/forte/image_annotation_test.py:96]: I guess it's the same as `Box.init_from_center_n_shape`

File: forte/data/ontology/top.py (hunk 3 above)
Comment (hepengfe) [reply forte/data/ontology/top.py:1264]: I haven't thought of an actually difference between `BoundingBox` and `Box` yet. Previously, `BoundingBox` requires `Grid` to function correctly but it doesn't seem to be the case. I am thinking we can keep it here and we can add more attributes/functions as needed later.

File: tests/forte/image_annotation_test.py (hunk 4 above)
Comment (hepengfe) [reply tests/forte/image_annotation_test.py:96]: It shows the child class can init from center and shape as well, and it keeps the child class type.

File: forte/data/ontology/top.py [hunk 5]
```diff
@@ -1067,33 +1067,118 @@ class Box(Region):
...
         image_payload_idx: int = 0,
     ):
-        # assume Box is associated with Grids
-        super().__init__(pack, image_payload_idx)
+        """
+        A class method to initialize a ``Box`` from a box's center position and
+        shape.
+
+        Args:
+            pack: the container that this ``BoundingBox`` will be added to.
```
[the remark this answers is not in the document]
Comment (hepengfe) [reply forte/data/ontology/top.py:1130]: Corrected it.

**Comment by hepengfe** [comment]
Actually, not one of those PR require `BoundingBox` so I think we can still include here to differentiate `BoundingBox` and `Box`. And current  `BoundingBox` is in vanilla status and can be added more functionality as needed.

File: forte/data/ontology/top.py [hunk 6]
```diff
@@ -1105,26 +1190,27 @@ def corners(self):
...
 
     @property
     def box_min_x(self):
-        return max(self._cx - round(0.5 * self._width), 0)
+        return max(self.x0, 0)
 
     @property
     def box_max_x(self):
-        return min(self._cx + round(0.5 * self._width), self.max_x)
+        return min(self.x1, self.pack.image_width)
```
[the remark this answers is not in the document]
Comment (hepengfe) [reply forte/data/ontology/top.py]: it's actually a mistake and I have corrected it.

File: forte/data/ontology/top.py [hunk 7]
```diff
@@ -1105,26 +1190,27 @@ def corners(self):
...
+            (self.y0, self.x0),
+            (self.y0, self.x1),
+            (self.y1, self.x0),
+            (self.y1, self.x1),
         ]
 
     @property
     def box_min_x(self):
-        return max(self._cx - round(0.5 * self._width), 0)
+        return max(self.x0, 0)
```
[the remark this answers is not in the document]
Comment (hepengfe) [reply forte/data/ontology/top.py]: removed the comparison

File: forte/data/ontology/top.py (hunk 3 above)
Comment (hepengfe) [reply forte/data/ontology/top.py]: Removed and it's ready for review now.

**Review by mylibrar — APPROVED** [review]

**Pull request merged**